library/core/src/str/mod.rs

```diff
@@ -758,7 +762,9 @@ impl str {
     #[inline]
     #[must_use]
     #[stable(feature = "split_at_checked", since = "1.80.0")]
-    pub fn split_at_mut_checked(&mut self, mid: usize) -> Option<(&mut str, &mut str)> {
+    #[rustc_const_unstable(feature = "const_str_split_at", issue = "131518")]
+    #[rustc_allow_const_fn_unstable(const_is_char_boundary)]
```

Comment:
Why did you add `rustc_allow_const_fn_unstable` here? That makes no sense on an unstable const fn.

Also @Noratrieb please make sure to get wg-const-eval approval before adding any rustc_allow_const_fn_unstable.

Reply:
Ideally we'd have a bot helping with this attribute policy... https://github.com/rust-lang/triagebot/issues/1851

Reply:
ah, I didn't see this, yeah. I was a bit unsure whether I need to ping const eval here but I deemed that there wasn't anything weird enough going on to be problematic (which I guess is sort of true here, the attribute is just misplaced). will be more careful next time

Reply:
> Why did you add `rustc_allow_const_fn_unstable` here? That makes no sense on an unstable const fn. (I'm cleaning this up in #132451.)

I added it because I got an error oyherwise, though that was on an earlier version of the PR.